src/core/server/services/slack/publisher.ts

```diff
@@ -0,0 +1,214 @@
+import { CommentCreatedInput } from "coral-server/graph/tenant/resolvers/Subscription/commentCreated";
+import { CommentEnteredModerationQueueInput } from "coral-server/graph/tenant/resolvers/Subscription/commentEnteredModerationQueue";
+import { CommentFeaturedInput } from "coral-server/graph/tenant/resolvers/Subscription/commentFeatured";
+import { CommentLeftModerationQueueInput } from "coral-server/graph/tenant/resolvers/Subscription/commentLeftModerationQueue";
+import { CommentReleasedInput } from "coral-server/graph/tenant/resolvers/Subscription/commentReleased";
+import { CommentReplyCreatedInput } from "coral-server/graph/tenant/resolvers/Subscription/commentReplyCreated";
+import { CommentStatusUpdatedInput } from "coral-server/graph/tenant/resolvers/Subscription/commentStatusUpdated";
+import { SUBSCRIPTION_CHANNELS } from "coral-server/graph/tenant/resolvers/Subscription/types";
+import logger from "coral-server/logger";
+
+import { GQLMODERATION_QUEUE } from "coral-server/graph/tenant/schema/__generated__/types";
+
+import { Tenant } from "coral-server/models/tenant";
+import { Db } from "mongodb";
+import SlackContext from "./context";
+
+type Payload =
+  | CommentEnteredModerationQueueInput
+  | CommentLeftModerationQueueInput
+  | CommentStatusUpdatedInput
+  | CommentReplyCreatedInput
+  | CommentCreatedInput
+  | CommentFeaturedInput
+  | CommentReleasedInput;
+
+function enteredModeration(channel: SUBSCRIPTION_CHANNELS, payload: Payload) {
+  const p: any = payload;
+  return (
+    channel === SUBSCRIPTION_CHANNELS.COMMENT_ENTERED_MODERATION_QUEUE &&
+    p.hasOwnProperty("queue") &&
+    (p.queue === GQLMODERATION_QUEUE.REPORTED ||
+      p.queue === GQLMODERATION_QUEUE.PENDING)
+  );
+}
+
+function isFeatured(channel: SUBSCRIPTION_CHANNELS) {
+  return channel === SUBSCRIPTION_CHANNELS.COMMENT_FEATURED;
+}
+
+function isReported(channel: SUBSCRIPTION_CHANNELS, payload: Payload) {
+  const p: any = payload;
+  return (
+    channel === SUBSCRIPTION_CHANNELS.COMMENT_ENTERED_MODERATION_QUEUE &&
+    p.hasOwnProperty("queue") &&
+    p.queue === GQLMODERATION_QUEUE.REPORTED
+  );
+}
+
+function isPending(channel: SUBSCRIPTION_CHANNELS, payload: Payload) {
+  const p: any = payload;
+  return (
+    channel === SUBSCRIPTION_CHANNELS.COMMENT_ENTERED_MODERATION_QUEUE &&
+    p.hasOwnProperty("queue") &&
+    p.queue === GQLMODERATION_QUEUE.PENDING
+  );
+}
+
+function createModerationLink(rootURL: string, commentID: string) {
+  return `${rootURL}/admin/moderate/comment/${commentID}`;
+}
+
+function createCommentLink(storyURL: string, commentID: string) {
+  return `${storyURL}?commentID=${commentID}`;
```

Comment:
As with `src/core/server/services/users/download/download.ts:107`, you should modify this to support story URL's that contain query params:

```suggestion
  const url = new URL(storyURL);
  url.searchParams.set('commentID', commentID);
  return url.href;
```

`URL` is from `import { URL } from 'url';`.